File: Integraupt-Backend/BackendAdReserva/src/main/java/com/AdReserva/dto/AdminReservaFilter.java
package com.AdReserva.dto;

import java.time.LocalDate;
import java.util.Optional;

public record AdminReservaFilter(
        String estado,
        String tipoEspacio,
        Integer facultadId,
        Integer escuelaId,
        LocalDate fechaReserva,
        String search,
        Integer usuarioId,
        String rol
) {
    public Optional<String> estadoOpt() {
        return Optional.ofNullable(estado).filter(value -> !value.isBlank());
    }

    public Optional<String> tipoEspacioOpt() {
        return Optional.ofNullable(tipoEspacio).filter(value -> !value.isBlank());
    }

    public Optional<Integer> facultadIdOpt() {
        return Optional.ofNullable(facultadId);
    }

    public Optional<Integer> escuelaIdOpt() {
        return Optional.ofNullable(escuelaId);
    }

    public Optional<LocalDate> fechaReservaOpt() {
        return Optional.ofNullable(fechaReserva);
    }

    public Optional<String> searchOpt() {
        return Optional.ofNullable(search).filter(value -> !value.isBlank());
    }

    public Optional<Integer> usuarioIdOpt() {
        return Optional.ofNullable(usuarioId);
    }

    public Optional<String> rolOpt() {
        return Optional.ofNullable(rol)
                .map(String::trim)
                .filter(value -> !value.isEmpty());
    }
}

File: Integraupt-Backend/BackendAdReserva/src/DTO/AdminReservaFilter.php
<?php

namespace App\DTO;

class AdminReservaFilter {
    public ?string $estado;
    public ?string $tipoEspacio;
    public ?int $facultadId;
    public ?int $escuelaId;
    public ?string $fechaReserva; // Y-m-d
    public ?string $search;
    public ?int $usuarioId;
    public ?string $rol;

    public function __construct(
        ?string $estado,
        ?string $tipoEspacio,
        ?int $facultadId,
        ?int $escuelaId,
        ?string $fechaReserva,
        ?string $search,
        ?int $usuarioId,
        ?string $rol
    ) {
        $this->estado = $estado;
        $this->tipoEspacio = $tipoEspacio;
        $this->facultadId = $facultadId;
        $this->escuelaId = $escuelaId;
        $this->fechaReserva = $fechaReserva;
        $this->search = $search;
        $this->usuarioId = $usuarioId;
        $this->rol = $rol;
    }

    public function getEstado(): ?string {
        return $this->estado !== null && trim($this->estado) !== '' ? trim($this->estado) : null;
    }

    public function getTipoEspacio(): ?string {
        return $this->tipoEspacio !== null && trim($this->tipoEspacio) !== '' ? trim($this->tipoEspacio) : null;
    }

    public function getFacultadId(): ?int {
        return $this->facultadId;
    }

    public function getEscuelaId(): ?int {
        return $this->escuelaId;
    }

    public function getFechaReserva(): ?string {
        return $this->fechaReserva !== null && trim($this->fechaReserva) !== '' ? trim($this->fechaReserva) : null;
    }

    public function getSearch(): ?string {
        return $this->search !== null && trim($this->search) !== '' ? trim($this->search) : null;
    }

    public function getUsuarioId(): ?int {
        return $this->usuarioId;
    }

    public function getRol(): ?string {
        return $this->rol !== null && trim($this->rol) !== '' ? trim($this->rol) : null;
    }
}
